stack.py

```diff
@@ -112,11 +127,16 @@ def eval_program(program):
             if dup_depth > len(stack):
                 raise IndexError
             stack.extend(stack[-dup_depth:])
+        elif instr.op == 'not':
```

Comment:
You should probably generalize this to `is_unary_operator` sooner rather than later, just for uniformity's sake.